internal/providers/terraform/dir_provider.go

```diff
@@ -470,39 +474,48 @@ func checkTerraformVersion(v string, fullV string) error {
 	return nil
 }
 
-func (p *DirProvider) printTerraformErr(err error) {
+func (p *DirProvider) buildTerraformErr(err error) error {
 	stderr := extractStderr(err)
-	if stderr == "" {
-		return
-	}
 
 	binName := "Terraform"
 	if p.IsTerragrunt {
 		binName = "Terragrunt"
 	}
 
-	msg := fmt.Sprintf("\n  %s command failed with:\n%s\n", binName, ui.Indent(stderr, "    "))
+	msg := ""
 
-	if strings.HasPrefix(stderr, "Error: Failed to select workspace") {
-		msg += "\nRun `terraform workspace select your_workspace` first or set the TF_WORKSPACE environment variable.\n"
+	if stderr != "" {
+		msg += fmt.Sprintf("\n\n  %s command failed with:\n%s", binName, ui.Indent(stderr, "    "))
 	}
-	if strings.HasPrefix(stderr, "Error: Required token could not be found") {
-		msg += "\nRun `terraform login` first or set the TF_CLI_CONFIG_FILE environment variable to the ABSOLUTE path.\n"
-	}
-	if strings.HasPrefix(stderr, "Error: No value for required variable") {
-		msg += "\nPass Terraform flags using the --terraform-plan-flags option.\n"
-		msg += "For example: infracost --path=path/to/terraform --terraform-plan-flags=\"-var-file=my.tfvars\"\n"
-	}
-	if strings.HasPrefix(stderr, "Error: Failed to read variables file") {
-		msg += "\nSpecify the -var-file flag as a path relative to your Terraform directory.\n"
-		msg += "For example: infracost --path=path/to/terraform --terraform-plan-flags=\"-var-file=my.tfvars\"\n"
-	}
-	if strings.HasPrefix(stderr, "Error: error configuring Terraform AWS Provider: no valid credential sources for Terraform AWS Provider found.") {
-		msg += "\nTerraform requires AWS credentials to be set.\n"
-		msg += "For more details, see https://registry.terraform.io/providers/hashicorp/aws\n"
+
+	if strings.HasPrefix(stderr, "Error: Failed to select workspace") {
+		msg += "\n\nYou have two options:\n"
+		msg += "1. Run `terraform workspace select your_workspace` first or set the TF_WORKSPACE environment variable.\n\n"
+		msg += fmt.Sprintf("2. Set --path to a Terraform plan JSON file. See %s for how to generate this.", ui.LinkString("https://infracost.io/troubleshoot"))
+	} else if errors.Is(err, ErrMissingCloudToken) || errors.Is(err, ErrInvalidCloudToken) || strings.HasPrefix(stderr, "Error: Required token could not be found") {
```

Comment:
I merged these errors into a single message, since they should follow the same solution path.